File: php/features.php
<?php
declare(strict_types=1);

// Brasil SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BrasilFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BrasilBaseFeature();
            case "test":
                return new BrasilTestFeature();
            default:
                return new BrasilBaseFeature();
        }
    }
}

File: py/features.py
# Brasil SDK feature factory

from feature.base_feature import BrasilBaseFeature
from feature.test_feature import BrasilTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BrasilBaseFeature(),
        "test": lambda: BrasilTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
